zebra-state/src/service.rs

```diff
@@ -252,6 +252,87 @@ impl StateService {
     pub fn utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Output> {
         self.mem.utxo(outpoint).or_else(|| self.sled.utxo(outpoint))
     }
+
+    /// Return an iterator over the relevant chain of the block identified by
+    /// `hash_or_height`.
+    #[allow(dead_code)]
+    pub fn chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: State::NonFinalized(hash_or_height),
+        }
+    }
+}
+
+struct Iter<'a> {
+    service: &'a StateService,
+    state: State,
+}
+
+enum State {
+    NonFinalized(HashOrHeight),
+    Finalized(block::Height),
+    Finished,
+}
+
+impl<'a> Iter<'a> {
+    fn next_non_finalized_block(&mut self) -> Option<Arc<Block>> {
+        let Iter { service, state } = self;
+
+        let hash_or_height = match state {
+            State::NonFinalized(hash_or_height) => hash_or_height,
+            State::Finalized(_) | State::Finished => unreachable!(),
+        };
+
+        if let Some(block) = service.mem.block(*hash_or_height) {
```

Comment:
`block` returns the block in the current best chain, but `chain` needs to return side-chain blocks, if it's given a side-chain hash.